pkg/trace/config/config.go

```diff
@@ -111,8 +111,81 @@ type AgentConfig struct {
 	Obfuscation *ObfuscationConfig
 }
 
+// setKnownKeys sets all the "root" keys known by the trace-agent. If any are missing,
+// the agent will log a warning message when loading the configuration:
+//   Unknown key in config file: <key_name>
+func setKnownKeys() {
+	config.Datadog.SetKnown("apm_config.enabled")
+	config.Datadog.SetKnown("apm_config.env")
+	config.Datadog.SetKnown("apm_config.additional_endpoints")
+	config.Datadog.SetKnown("apm_config.apm_non_local_traffic")
+	config.Datadog.SetKnown("apm_config.max_traces_per_second")
+	config.Datadog.SetKnown("apm_config.max_memory")
+	config.Datadog.SetKnown("apm_config.log_file")
+	config.Datadog.SetKnown("apm_config.apm_dd_url")
+	config.Datadog.SetKnown("apm_config.max_cpu_percent")
+	config.Datadog.SetKnown("apm_config.receiver_port")
+	config.Datadog.SetKnown("apm_config.connection_limit")
+	config.Datadog.SetKnown("apm_config.ignore_resources")
+	config.Datadog.SetKnown("apm_config.replace_tags")
+	config.Datadog.SetKnown("apm_config.obfuscation")
+	config.Datadog.SetKnown("apm_config.obfuscation.elasticsearch")
+	config.Datadog.SetKnown("apm_config.obfuscation.elasticsearch.enabled")
+	config.Datadog.SetKnown("apm_config.obfuscation.elasticsearch.keep_values")
+	config.Datadog.SetKnown("apm_config.obfuscation.mongodb")
+	config.Datadog.SetKnown("apm_config.obfuscation.mongodb.enabled")
+	config.Datadog.SetKnown("apm_config.obfuscation.mongodb.keep_values")
+	config.Datadog.SetKnown("apm_config.obfuscation.http")
+	config.Datadog.SetKnown("apm_config.obfuscation.http.remove_query_string")
+	config.Datadog.SetKnown("apm_config.obfuscation.http.remove_paths_with_digits")
+	config.Datadog.SetKnown("apm_config.obfuscation.remove_stack_traces")
+	config.Datadog.SetKnown("apm_config.obfuscation.redis")
+	config.Datadog.SetKnown("apm_config.obfuscation.redis.enabled")
+	config.Datadog.SetKnown("apm_config.obfuscation.memcached")
+	config.Datadog.SetKnown("apm_config.obfuscation.memcached.enabled")
+	config.Datadog.SetKnown("apm_config.extra_sample_rate")
+	config.Datadog.SetKnown("apm_config.dd_agent_bin")
+	config.Datadog.SetKnown("apm_config.max_events_per_second")
+	config.Datadog.SetKnown("apm_config.trace_writer")
```

Comment:
I think specifying the parent key explicitly is not needed if you specify a subkey.

Reply:
Cool! I removed parents.